tests/test_scripts.py

```diff
@@ -82,3 +84,75 @@ def test_proto_conversion(tmpdir):
     assert p.returncode == 0
 
     assert p.stdout.startswith(b'{\n  "meta": ') or p.stdout.startswith(b'{\r\n  "meta": ')
+
+
+def run_detect_duplicate_features(rule_dir, rule_path):
+    # rule_path = "collection/credit-card/parse-credit-card-information.yml"
+    args = [rule_dir, rule_path]
+    script_path = get_script_path("detect_duplicate_features.py")
+    args = [sys.executable] + [script_path] + args
+    print(f"running: '{args}'")
+    return subprocess.run(args)
+
+
+def test_detect_duplicate_features(tmpdir):
+    RULESET = {
+        "rule_1": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 1
+                        scope: function
+                    features:
+                      - or:
+                        - string: "sites.ini"
+                        - number: 0xEDB88320
+            """
+        ),
+        "rule_2": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 2
+                        scope: function
+                    features:
+                      - and:
+                        - string: "sites.ini"
+                        - number: 8
+            """
+        ),
+        "rule_3": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 3
+                        scope: function
+                    features:
+                      - not:
+                        - number: 0xEDB88320
+            """
+        ),
+        "rule_4": textwrap.dedent(
+            """
+                rule:
+                    meta:
+                        name: Test Rule 4
+                        scope: function
+                    features:
+                      - not:
+                        - number: 4
+            """
+        ),
+    }
+
+    rule_dir = tmpdir.mkdir("capa_rule_overlap_test")
+    rule_overlaps = [3, 2, 2, 1]
```

Comment:
it takes a while to see what this means/why these numbers work, can you add a comment which overlaps are present here?

Reply:
Thanks, for the details. Let's move the comment before the definition.
I think it would be a bit more intuitive (and how we would use to script) to have an existing rule set and compare the overlap to a new rule that's outside of that set. Then we can have different overlap counts between existing ruleN and the new rule (including 1 overlap, multiple overlaps, and 0 overlaps).

Reply:
0 overlaps are not possible, since the rule we are checking is already present in RULESET

Reply:
Correct, that's why I proposed: "I think it would be a bit more intuitive (and how we would use to script) to have an existing rule set and compare the overlap to a new rule that's outside of that set."